File: src/main/java/io/gaegul/ch01/puzzle/King.java
package io.gaegul.ch01.puzzle;

import io.gaegul.ch01.puzzle.behavior.SwordBehavior;

public class King extends Character {

	protected King() {
		super(new SwordBehavior());
	}

	@Override
	public void fight() {
		System.out.println("왕이 전투합니다.");
	}
}

File: src/main/kotlin/io/gaegul/ch01/puzzle/King.kt
package io.gaegul.ch01.puzzle

import io.gaegul.ch01.puzzle.behavior.SwordBehavior

class King: Character(
    SwordBehavior()
) {

    override fun fight() {
        print("왕은 ")
        weaponBehavior.useWeapon()
        print("으로 전투할 수 있다.")
    }
}